### PR TITLE
Fix WinHttpHandler default cred auth when server offers multiple schemes

### PR DESCRIPTION
**Pull request by davidsh**
When a server offers multiple auth schemes, WinHttpHandler is supposed
to select the more secure scheme regardless of the order of the
authentication headers. So, despite a server offering Basic and
Negotitate (in that order), Negotiate would be selected.

When using default credentials, however, it was not calling
WinHttpSetCredentials(null, null). This resulted in the first scheme
offered by the server (Basic) to be attempted. But there were no
credentials set and so a 401 resulted.

Fix WinHttpHandler so that it will always call WinHttpSetCredentials
even with default credentials. This ensures that the selected auth
scheme (in this case Negotiate) is used correctly.

Revised test configuration instructions. An IHttpModule is used for the
test server endpoint to ensure that IIS will put 'Basic' first in the
response headers.

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @stephentoub @CIPop

File: src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs [hunk 1]
```diff
@@ -302,6 +302,10 @@ public void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
             uint authScheme,
             uint authTarget)
         {
+            string userName;
+            string password;
+            string domain;
```
Comment (stephentoub) [inline src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs]: Does ```domain``` need to be declared here?  It seems like it's only used inside the below else block.

**Comment by stephentoub** [comment]
LGTM

File: src/System.Net.Http/tests/FunctionalTests/prerequisites/readme.txt [hunk 2]
```diff
@@ -3,7 +3,14 @@
...
-Set up IIS and ASP.NET on the test server. Create an application directory with the necessary path (see DefaultCredentialsTest.cs).
-The directory should only have Integrated Windows authentication access enabled. Disable Anonymous access.
+Set up IIS and ASP.NET on the test server. Create two application directories with the following paths:
+
+/test/auth/negotiate
+/test/auth/multipleschemes
+
+Set authentication on both application directories to use integrated Windows authentication. Disable Anonymous access.
+
+In the 'mutlipleschemes' application, you'll use the BasicAuthModule.cs and web.config files in a compiled ASP.NET project. This allows Basic auth to
```
Comment (danmosemsft) [inline src/System.Net.Http/tests/FunctionalTests/prerequisites/readme.txt]: Typo in multiple.

File: src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs]: Moved inside the else block.

File: src/System.Net.Http/tests/FunctionalTests/prerequisites/readme.txt (hunk 2 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/prerequisites/readme.txt]: Thanks! Fixed.

**Pull request merged**